### ISSUE TITLE
dataclasses I3Orientation python printing fails (Trac #2182)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2174]
when printing an I3Orientation directly in python, using  print(str(<I3Orientation>))
it prints any orientation like this:

I3Orientation:
    Dir: (6.94834e-310,6.94834e-310,1.64477e-316)
     Up: (6.95303e-310,6.94833e-310,6.94833e-310)
  Right: (0,6.94832e-310,0)

so with only zeros in each component of the directions.
I'm using the combo stable.

(this is my first ticket, so if I forgot something important just ask; I don't seem to be able to log in as something else than icecube, my user name is elohfink)

Migrated from https://code.icecube.wisc.edu/ticket/2182
```json
{
    "status": "closed", 
    "changetime": "2018-08-23T18:33:34", 
    "description": "when printing an I3Orientation directly in python, using  print(str(<I3Orientation>))\nit prints any orientation like this:\n\nI3Orientation:\n    Dir: (6.94834e-310,6.94834e-310,1.64477e-316)\n     Up: (6.95303e-310,6.94833e-310,6.94833e-310)\n  Right: (0,6.94832e-310,0)\n\nso with only zeros in each component of the directions.\nI'm using the combo stable.\n\n(this is my first ticket, so if I forgot something important just ask; I don't seem to be able to log in as something else than icecube, my user name is elohfink)", 
    "reporter": "icecube", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1535049214287878", 
    "component": "combo core", 
    "summary": "dataclasses I3Orientation python printing fails", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-08-23T15:31:03", 
    "milestone": "", 
    "owner": "cweaver", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180823T15:42:56: cweaver commented:

> It looks like I may have overlooked that I3Orientation is doing sneaky tricks with calculating some of its members on the fly, so creating and then printing one with no other operations between might be printing uninitialized garbage.

**Comment by icecube-trac** [comment]
Trac update at 20180823T18:33:34: 
* cweaver changed owner from "" to "cweaver"
* cweaver commented:

> In changeset 164967:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="164967"
> Ensure that members are calculated before printing. Fixes #2174.
> ```


* cweaver changed resolution from "" to "fixed"
* cweaver changed status from "new" to "closed"
